pkg/controller/migplan/validation.go

```diff
@@ -448,3 +466,38 @@ func (r ReconcileMigPlan) validateDestinationNamespaces(plan *migapi.MigPlan) er
 
 	return nil
 }
+
+// Validate PV actions.
+func (r ReconcileMigPlan) validatePvAction(plan *migapi.MigPlan) error {
+	list := []string{}
+	actions := map[string]bool{
+		migapi.PvCopyAction: true,
+		migapi.PvMoveAction: true,
+	}
+	for _, pv := range plan.Spec.PersistentVolumes.List {
+		_, found := actions[pv.Action]
+		if !found {
+			list = append(list, pv.Name)
+		}
+	}
+	if len(list) > 0 {
+		choices := []string{}
+		for key := range actions {
+			choices = append(choices, key)
+		}
+		message := fmt.Sprintf(
+			PvInvalidActionMessage,
+			strings.Join(list, ", "),
+			strings.Join(choices, "|"))
+		plan.Status.SetCondition(migapi.Condition{
```

Comment:
Are we setting one condition for each invalid PV choice here?

Reply:
No, it's a single condition listing all the PVs with invalid actions.